mesh/pmesh.cpp

```diff
@@ -1594,6 +1594,15 @@ void ParMesh::SetAttributes()
    }
 }
 
+bool ParMesh::HasBoundaryElements() const
+{
+   // maximum number of boundary elements over all ranks
+   int maxNumOfBdrElements;
+   MPI_Allreduce(&NumOfBdrElements, &maxNumOfBdrElements, 1,
+                 MPI_C_BOOL, MPI_MAX, MyComm);
```

Comment:
It turns out `MPI_MAX` is not supported for `MPI_C_BOOL`, we will need to change the type:

```
MPI_Allreduce: the reduction operation MPI_MAX is not defined on the MPI_C_BOOL datatype
```

Reply:
That's a bug, it should have been `MPI_INT`

Reply:
I added the fix to this branch and `master`